File: alogrytm/silnia.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  potega.py
#
# Obliczanie silni liczby naturlnej

def silnia_it(n):
    """
    n! = 1 * 2 * ... * n
    """
    wynik = 1
    
    for i in range(n):
        wynik = wynik * i 
        
    
    return wynik
    
def potenga2_it(a, n):
    
    print(a**n)
    
    
def main(args):
    n = int(input("Podaj silnie: "))
    
    print("{}! silnia wynosi {}".
            format(n, silnia_it(n)))
    # potenga_it(a, n)    # Wywołanie funkcji
    


    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: CPP/silnia.cpp
/*
 * silnia.cpp
 */


#include <iostream>
using namespace std;

int silnia_it(int a){
    int wynik = 1;
    
    for(int i = 1; i <= a; i++){
        wynik = wynik * i; 
        
    }
    return wynik;
}

int silnia_re(int a)
{
    if (a == 0) return 1;  
    return silnia_re(a-1) * a;  
}

int main(int argc, char **argv)
{
    int a;
    cout << "Podaj liczbę ";
    cin >> a;
    cout << "Silnia: " << silnia_it(a) << endl;
    cout << "Silnia: " << silnia_re(a) << endl;
    return 0;
}

